template/interpolate/aws/secretsmanager/secretsmanager.go

```diff
@@ -96,13 +97,24 @@ func getSecretValue(s *SecretString, spec *SecretSpec) (string, error) {
 
 	if spec.Key == "" {
 		for _, v := range secretValue {
-			return v, nil
+			return getStringSecretValue(v)
 		}
 	}
 
 	if v, ok := secretValue[spec.Key]; ok {
-		return v, nil
+		return getStringSecretValue(v)
 	}
 
 	return "", fmt.Errorf("No secret found for key %q", spec.Key)
 }
+
+func getStringSecretValue(v interface{}) (string, error) {
+	switch valueType := v.(type) {
+	case string:
+		return v.(string), nil
+	case float64:
+		return strconv.FormatFloat(v.(float64), 'f', 0, 64), nil
+	default:
+		return "", fmt.Errorf("Unsupported secret value type: %v", valueType)
+	}
+}
```

Comment:
Here the type switch returns something of the type you want:
```suggestion
func getStringSecretValue(v interface{}) (string, error) {
	switch valueType := v.(type) {
	case string:
		return valueType, nil
	case float64:
		return strconv.FormatFloat(valueType, 'f', 0, 64), nil
	default:
		return "", fmt.Errorf("Unsupported secret value type: %v", valueType)
	}
}
```